design-documents/frontend/technical-vision.md

```diff
@@ -1,144 +1,185 @@
+1. Build upon step 1 with JavaScript behaviors, taking care to preserve the HTML you have already created. Remember that most of the behavior you need is already provided by the HTML forms &ndash; less is more! At this stage, you should be mindful of what classes need to be added to certain elements in preparation for a style pass.
+1. Styles &ndash; do your best to _not_ touch the HTML or JavaScript at this stage. Use your CSS wizardry to achieve the design specifications provided to you. Do everyone a favor and provide screenshots of the final outcome in your pull request so they know what they are reviewing.
+
+## Code Quality
+
+_Quality, Maintainability, Productivity._
+
+We should leverage the built-in code coverage reports that [Jest][] provides, but also the following GitHub integrations/apps:
+
+- Continuous integration with [Travis-CI](https://travis-ci.com/).
+- [CodeClimate](https://codeclimate.com/)
+  - [**Quality** by Code Climate](https://codeclimate.com/quality/) provides automated code review for test coverage, **maintainability** and more so that you can **save time** and merge with confidence.
+  - Code Climate’s engineering process insights and automated code review for GitHub and GitHub Enterprise help you ship **better software, faster**.
+
+### Resources
+- [Code Climate for TypeScript is here!](https://codeclimate.com/changelog/5a147fcac7d08102b700084e/)
+- [Code Climate | TSLint](https://docs.codeclimate.com/docs/tslint)
+
+## Testing
+
+There are many types of tests you can have for an application and many variables that ultimately determine how much value you give to each one. [The Testing Tropy](https://testingjavascript.com/) provides a good visualization for both the importance/size and the placement of each type.
+
+[<img src="https://res.cloudinary.com/dg3gyk0gu/image/upload/v1539186394/theTestingTrophy_2x.png" alt="The Testing Trophy" height="373">](https://testingjavascript.com/)
+
+### Static Type System with [TypeScript][]
+
+When building an application, like building a trophy, we start at the foundation with a static type system and linter (e.g., [TypeScript][] and [TSLint][]). This sets up a good foundation and comes at a relatively low cost for new projects.
+
+### Unit & Integration Tests with [Jest][]
+
+The next layer of this trophy is built with unit tests. With [Jest][], these also come at a relatively low cost, because of how simple it is to write snapshot tests. These are especially useful when testing [React][] components with the speed and efficiency of shallow rendering &ndash; a cheap win!
```

Comment:
> These are especially useful when testing [React][] components with the speed and efficiency of shallow rendering &ndash; a cheap win!

You recommend the `testingjavascript.com` course materials, but these (and the author) advise against shallow rendering - seems confusing.

Reply:
Resolved [here](https://github.com/magento/architecture/pull/67/commits/18397d204f1e90a43700f2dc332f68a8d66e21a2#diff-13955aa5a2f331269e98555b510dc546R164).